src/dcm_classifier/utility_functions.py

```diff
@@ -451,20 +451,31 @@ def sanitize_dicom_dataset(
 
     # set the default values for optional dicom fields
     for field in optional_info_list:
+        DEFAULT_VALUE = -12345
+        dataset_dictionary[field] = (
+            DEFAULT_VALUE  # Every optional field will be set to the default -12345 and will be overriden if present and valid
+        )
+        try:
+            dataset_value = dataset[field].value
+        except Exception:
+            dataset_value = DEFAULT_VALUE
+
         if field == "SAR":
-            if field not in dataset or not is_number(dataset[field].value):
+            if field not in dataset or not is_number(dataset_value):
                 # SAR is allowed to be empty or not a number because derived images often do not have SAR
                 # for example, ADC images are derived images that are computed, so there is not
                 # SAR impact on the patient for the derived image.
                 # SAR Calculated whole body Specific Absorption Rate in watts/kilogram.
                 # indicate that there is no SAR for the computed image
-                _default_inferred_value = -12345.0
+                _default_inferred_value = DEFAULT_VALUE
                 dataset_dictionary[field] = _default_inferred_value
                 vprint(
                     f"Inferring optional {field} value of '{_default_inferred_value}' for missing field in {dicom_filename}"
                 )
             else:
-                dataset_dictionary[field] = dataset[field].value
+                dataset_dictionary[field] = validate_numerical_dataset_element(
+                    dataset_value
+                )
         elif field == "Manufacturer":
```

Comment:
for fields that do not use the new "validate_numerical_dataset_element" function, ensure that they are sanitized for the None Type